app/controllers/use_case_controller.rb

```diff
@@ -0,0 +1,74 @@
+# Abstract controller – requires the following methods on the subclass:
+# * model_class
+# * permitted_params
+class UseCaseController < ApplicationController
+  before_filter :authenticate_user!
+  before_filter ->{authorize self.proposal}, only: [:edit, :update]
+  rescue_from Pundit::NotAuthorizedError, with: :auth_errors
+  before_filter :find_model_instance, only: [:edit, :update]
```

Comment:
Nah, keeping consistent authorization commands across controllers seems more important -- we can probably even pull those out into the base controller/trait.